kroxylicious-kms-provider-fortanix-dsm/src/test/java/io/kroxylicious/kms/provider/fortanix/dsm/FortanixDsmKmsEdekSerdeTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.fortanix.dsm;
+
+import java.nio.ByteBuffer;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import io.kroxylicious.kms.service.Serde;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class FortanixDsmKmsEdekSerdeTest {
+
+    private static final String KEY_REF = "1234abcd-12ab-34cd-56ef-1234567890ab";
+    private static final byte[] EDEK = { 1, 2, 3 };
+    static final byte[] IV = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 };
+    static final byte[] ANOTHER_IV = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 0 };
+    private final Serde<FortanixDsmKmsEdek> serde = FortanixDsmKmsEdekSerde.instance();
+
+    @Test
+    void shouldRoundTrip() {
```

Comment:
should we also do a `serialize` test that checks the bytes get serialized into the v0 format? The roundtrip is a good property to check but I think it's distinct from checking that the Serde writes version as a byte, then x, then y etc.